stl/inc/atomic

```diff
@@ -161,7 +161,11 @@ _Ty kill_dependency(_Ty _Arg) noexcept { // "magic" template that kills dependen
 // FUNCTION _Check_memory_order
 inline void _Check_memory_order(const memory_order _Order) noexcept {
     // check that _Order is a valid memory_order
+#if _HAS_CXX20
+    if (_Order > memory_order_seq_cst) {
+#else
     if (static_cast<unsigned int>(_Order) > memory_order_seq_cst) {
+#endif
         _INVALID_MEMORY_ORDER;

```

Comment:
I wonder whether it would be better to reimplement this function with a `switch` like the related functions below, to avoid this preprocessor logic. I suppose that the comparison emits better debug codegen, so no change requested - just an observation.